### ISSUE TITLE
Help - Inkbird IDT-34c-B

### ISSUE THREAD

**Issue by SerpikoIT** [1technophile/OpenMQTTGateway#2111]
good morning I have an inkbird idt-34c-b thermometer I wanted to know if it can be used with Openmqtt Gateway

**Comment by DigiH** [comment]
Hi @SerpikoIT 

As the IDT-34c-B seems to be a Bluetooth BBQ thermometer it really depends on it sending it temperature values through BLE advertising data.

Can you turn on **BT: Publish Advertisement data** in the gateways's HA UI and then post a few MQTT messages, along with which probes are connected and their temperatures.

![image](https://github.com/user-attachments/assets/0a52b066-64fb-4fb3-a967-adb9d875c35d)

If the relevant information is being broadcast in the BLE broadcasts, with your help  we can create a decoder for it, similar to the other BBQ thermometer decoders. But it might also be that the information can only be retrieved by connection to the device.

**Comment by SerpikoIT** [comment]
forgive me I don't know where to activate BT: Publish Advertisement data I have HA on raspberry

**Comment by DigiH** [comment]
In the HA discovered gateway UI, where all the other gateway settings can be adjusted.

Can't you see it in the **Overview** dashboard?

And just to confirm, you are using OpenMQTTGateway on an ESP32 and not possibly Theengs Gateway (HA Add-n)?

**Comment by SerpikoIT** [comment]
yes I just installed theeng gateway and configured yes I have OpenMQTTGateway, now what should I do??

**Comment by DigiH** [comment]
In the Theengs Gateway configuration page it is under **PUBLISH_ADVDATA**

<img width="741" alt="Screenshot 2024-11-13 at 09 57 06" src="https://github.com/user-attachments/assets/fdc70eb8-6ce7-4339-abe0-7dcba20c9ff5">

Turn it on and monitor the undecoded MQTT messages for your IDT-34c-B with MQTT Explorer and post some sample messages here with the information of which probes were connected and what temperatures in ºC it is showing on the device's display.

**Comment by SerpikoIT** [comment]
ok I'm not at home now, this afternoon I'll do everything and send you the images. Thank you

**Comment by SerpikoIT** [comment]
Here I was able to find this, I hope it helps
![bbq_nuovo](https://github.com/user-attachments/assets/70b30964-100f-4d32-b28c-d0ec68c4b5db)
![IMG_20241113_195010.jpg](https://github.com/user-attachments/assets/6aa61917-8612-470e-b572-8823aeb90b8f)

**Comment by DigiH** [comment]
Yes, this is what is required to see if the IDT-34c-B might be broadcasting its tempertures, and if it is possible to create a decoder for it. If you're sure this is the correct MAC address/id selected above.

Could you copy and past any messages, with the copy button next to the **Value** heading

![image](https://github.com/user-attachments/assets/8bdb7a7d-0efc-4a2e-9992-3a0b469b8ab0)

and then remove some of the probes and wait for the next MQTT message to copy, and take one probe into your hand to heat it up and again wait for the next message to copy.

With the 2-3 more examples it should be obvious, as currently I don't see any immediate similarities in the manufacturerdata with the display image showing almost four identical temperature values.

**Comment by SerpikoIT** [comment]
ok I'll try right away


{"id":"A4:C1:38:C6:EE:E7","name":"IDT-34c-B","rssi":-46}


![photo_2024-11-14_09-04-23](https://github.com/user-attachments/assets/f5d6b9d3-f978-4f0e-8d46-fde8edfccaf6)


{"id":"A4:C1:38:C6:EE:E7","name":"IDT-34c-B","rssi":-39}


![photo_2024-11-14_09-04-52](https://github.com/user-attachments/assets/b7eca57a-e347-4370-bd33-c1b2f21d3a62)


{"id":"A4:C1:38:C6:EE:E7","name":"IDT-34c-B","rssi":-41}


![photo_2024-11-14_09-06-49](https://github.com/user-attachments/assets/6a66bc65-91ab-4578-ac1f-e64a779f66bb)

{"id":"A4:C1:38:C6:EE:E7","name":"IDT-34c-B","rssi":-53}


![photo_2024-11-14_09-09-57](https://github.com/user-attachments/assets/77f52a11-d3a2-4fd7-a00d-146ba7a90f21)



{"id":"A4:C1:38:C6:EE:E7","name":"IDT-34c-B","rssi":-51}


![photo_2024-11-14_09-14-32](https://github.com/user-attachments/assets/94d9a63c-0e4f-4efd-8fc9-d287c38339ac)




I hope this goes well

**Comment by DigiH** [comment]
This would have been perfect, but the undecoded raw **manufacturerdata** is not included in the messages any longer 😞 

Did you turn off **BT: Publish Advertisement data**?

Any chance of the above replaced with the **manufacturerdata** included?

**Comment by SerpikoIT** [comment]
![Screenshot_2024-11-14-09-39-44-831_io.homeassistant.companion.android.jpg](https://github.com/user-attachments/assets/c3968e23-be8b-4bb2-a271-b9560e174769)

😞😞😞😞

**Comment by SerpikoIT** [comment]
{"manufacturerdata": "1f846266a4c138c6eee700", "name": "IDT-34c-B", "id": "A4:C1:38:C6:EE:E7", "rssi": -36}


![photo_2024-11-14_09-43-05](https://github.com/user-attachments/assets/ee270acf-70cf-4f36-8888-0d93f55f2b23)


{"manufacturerdata": "1f846266a4c138c6eee700", "name": "IDT-34c-B", "id": "A4:C1:38:C6:EE:E7", "rssi": -40}


![photo_2024-11-14_09-45-19](https://github.com/user-attachments/assets/a14d80a2-1c8d-4631-802c-fc883c3abd41)


{"manufacturerdata": "1f846266a4c138c6eee700", "name": "IDT-34c-B", "id": "A4:C1:38:C6:EE:E7", "rssi": -33}


![photo_2024-11-14_09-46-29](https://github.com/user-attachments/assets/37109b78-5609-4b21-aaee-1fdd9a438bc1)

**Comment by SerpikoIT** [comment]
MQTT_HOST | MQTT host address, i.e. the IP/hostname of your Moquitto Broker Add-on

should I put the HA IP?

**Comment by DigiH** [comment]
> {"manufacturerdata": "**1f846266a4c138c6eee700**", "name": "IDT-34c-B", "id": "A4:C1:38:C6:EE:E7", "rssi": -36}
> {"manufacturerdata": "**1f846266a4c138c6eee700**", "name": "IDT-34c-B", "id": "A4:C1:38:C6:EE:E7", "rssi": -40}
> {"manufacturerdata": "**1f846266a4c138c6eee700**", "name": "IDT-34c-B", "id": "A4:C1:38:C6:EE:E7", "rssi": -33}

Unfortunately the manufacturerdata doesn't change with the different probes attached or not and the different temperatures. Therefore it won't currently be possible to create a decoder through the BLE advertising data.

**Comment by SerpikoIT** [comment]
ok could you help me understand if the configuration of theengs is done correctly??? Does the HA IP address go to the MQTT Host part?

**Comment by DigiH** [comment]
> Does the HA IP address go to the MQTT Host part?

It depends on where you are running your MQTT Broker.

And **GENERAL_PRESENCE** should be off for you using it with HA, and **PRESENCE** also only if you are using a room presence implementation in HA.

**PUBLISH_ALL** and **PUBLISH_ADVDATA** can also be turned off again, now that you have posted the sample messages above.

**Comment by SerpikoIT** [comment]
ok thanks I'll do it now, I don't know whether to return my thermometer to Amazon or wait. I saw a guy say he was able to read the values ​​with catttool. I'll post the link so we'll see if he can help us out

https://community.home-assistant.io/t/how-to-tranform-gatttool-commands-into-availible-homeassistant-integration/642344

**Comment by DigiH** [comment]
It's very likely that you can get values by connection to the device, but that is out of the scope of Theengs Decoder which currently only decodes freely broadcast BLE advertising data.

You could try the [OpenMQTTGateway READ command](https://docs.openmqttgateway.com/use/ble.html#read-write-ble-characteristics-over-mqtt), to see if you can read from any appropriate service/characteristics.

**Comment by SerpikoIT** [comment]
I returned it to Amazon and then on Black Friday I'll get the other model if I can. Thank you

**Comment by DigiH** [comment]
@SerpikoIT Have a look at the already compatible devices/BBQ thermometers at

https://decoder.theengs.io/devices/devices.html
